File: schedulemodule/src/main/java/com/dxctraining/sm/entities/Schedule.java
package com.dxctraining.sm.entities;
import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Objects;

@Entity
public class Schedule {

    @Id
    @GeneratedValue
    private Integer scheduleId;
    //@OneToOne(targetEntity = Schedule.class)
    private String sourceAirport;
  //  @OneToOne(targetEntity = Schedule.class)
    private String destinationAirport;
    private LocalDateTime arrivalTime;
    private LocalDateTime departureTime;
    private String airportCode;

    public Schedule(String sourceAirport, String destinationAirport, LocalDateTime arrivalTime, LocalDateTime departureTime, String airportCode) {
        this.sourceAirport = sourceAirport;
        this.destinationAirport = destinationAirport;
        this.arrivalTime = arrivalTime;
        this.departureTime = departureTime;
        this.airportCode = airportCode;
    }

    public Schedule() {
    }

    public Integer getScheduleId() {
        return scheduleId;
    }

    public void setScheduleId(Integer scheduleId) {
        this.scheduleId = scheduleId;
    }

    public String getSourceAirport() {
        return sourceAirport;
    }

    public void setSourceAirport(String sourceAirport) {
        this.sourceAirport = sourceAirport;
    }

    public String getDestinationAirport() {
        return destinationAirport;
    }

    public void setDestinationAirport(String destinationAirport) {
        this.destinationAirport = destinationAirport;
    }

    public LocalDateTime getArrivalTime() {
        return arrivalTime;
    }

    public void setArrivalTime(LocalDateTime arrivalTime) {
        this.arrivalTime = arrivalTime;
    }

    public LocalDateTime getDepartureTime() {
        return departureTime;
    }

    public void setDepartureTime(LocalDateTime departureTime) {
        this.departureTime = departureTime;
    }

    public String getAirportCode() {
        return airportCode;
    }

    public void setAirportCode(String airportCode) {
        this.airportCode = airportCode;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Schedule schedule = (Schedule) o;
        return scheduleId.equals(schedule.scheduleId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(scheduleId);
    }
}

File: angular-integrated/src/app/model/schedule.ts
export class Schedule
{
    private sourceAirport:string;
    private destinationAirport:string;
    arrivalTime;
    departureTime;
    private airportCode:string;

    constructor(sourceAirport:string,destinationAirport:string,arrivalTime,departureTime,airportCode:string)
    {
            this.airportCode=airportCode;
            this.arrivalTime=arrivalTime;
            this.departureTime=departureTime;
            this.sourceAirport=sourceAirport;
            this.destinationAirport=destinationAirport;
    }
    getAirportCode():string{
        return this.airportCode;
    }
    getDepartureTime(){
        return this.departureTime
    }
    getArrivalTime()
    {
        return this.arrivalTime;
    }
    getsourceAirport():string{
        return this.sourceAirport;
    }
    getdestinationAirport():string{
        return this.destinationAirport;
    }
    

}